src/vaults/vaults.service.ts

```diff
@@ -33,6 +52,103 @@ export class VaultsService {
     return VaultsService.loadVault(chain, vaultDefinition, currency);
   }
 
+  async listVaultHarvests(chain: Chain): Promise<VaultHarvestsMap> {
+    const harvestsWithSnapshots = await Promise.all(
+      chain.vaults.map(async (vault) => {
+        return {
+          vault: vault.vaultToken,
+          harvests: await this.getVaultHarvests(chain, vault.vaultToken),
+        };
+      }),
+    );
+
+    return harvestsWithSnapshots.reduce((acc, harvestWithSnapshot) => {
+      acc[harvestWithSnapshot.vault] = harvestWithSnapshot.harvests;
+      return acc;
+    }, <VaultHarvestsMap>{});
+  }
+
+  async getVaultHarvests(chain: Chain, vaultAddr: VaultDefinition['vaultToken']): Promise<VaultHarvestsExtended[]> {
+    const vaultHarvests: VaultHarvestsExtended[] = [];
+
+    vaultAddr = ethers.utils.getAddress(vaultAddr.toLowerCase());
+
+    const sdk = await chain.getSdk();
+    const vaultDef = chain.vaults.find((vault) => vault.vaultToken === vaultAddr);
+
+    let sdkVaultHarvestsResp: {
+      data: VaultHarvestData[];
+    } = { data: [] };
+
+    if (!vaultDef) return vaultHarvests;
+
+    try {
+      sdkVaultHarvestsResp = await sdk.vaults.listHarvests({
+        address: vaultAddr,
+        version: vaultDef.version ?? VaultVersion.v1,
+      });
+    } catch (e) {
+      console.warn(`Failed to get harvests list ${e}`);
+    }
+
+    if (!sdkVaultHarvestsResp || sdkVaultHarvestsResp?.data?.length === 0) {
+      return vaultHarvests;
+    }
+
+    const sdkVaultHarvests = sdkVaultHarvestsResp.data;
+
+    const _extend_harvests_data = async (harvestsList: VaultPerformanceEvent[], eventType: HarvestType) => {
+      if (!harvestsList || harvestsList?.length === 0) return;
+
+      for (let i = 0; i < harvestsList.length; i++) {
+        const harvest = harvestsList[i];
+
+        const vaultGraph = await sdk.graph.loadSett({
+          id: vaultAddr,
+          block: { number: harvest.block },
+        });
+
+        const extendedHarvest = {
+          ...harvest,
+          eventType,
+          strategyBalance: 0,
+          estimatedApr: 0,
+        };
+
+        if (vaultGraph?.sett) {
+          extendedHarvest.strategyBalance = vaultGraph.sett?.strategy?.balance || 0;
+
+          if (i === harvestsList.length - 1) {
+            vaultHarvests.push(extendedHarvest);
+            continue;
+          }
+
+          const startOfHarvest = harvest.timestamp;
+          const endOfCurrentHarvest = harvestsList[i + 1].timestamp;
+
+          extendedHarvest.estimatedApr = await estimateHarvestEventApr(
```

Comment:
for this one we could probably drop the vault def and maybe should 
reason being is now there is `token` field on `VaultPerformanceEvent` 
the only time it is missing is for vault v1 harvests (present on distributions)

that is because it is implicitly the deposit token as you had in this method so you could just do like 

`event.token ?? vaultDef.depositToken`

Reply:
ohhh, this is wat i needed, thanks for explanation